src/main/java/soolutions/hometaskapp/core/Amount.java

```diff
@@ -0,0 +1,40 @@
+package soolutions.hometaskapp.core;
+
+// TODO(vivek): Write docs for each public class
+public final class Amount {
```

Comment:
I think `Amount` should be an interface as we've done with `Account`. For consistency's sake maybe implementing `Account with a `BasicAmount` class makes more sense.

Reply:
Also, I think maybe all of our domain objects should be defined in classes to encourage decoupling among implementations.

Reply:
Again i have to vote for all objects having the same interface

Reply:
@goqp can you justify your argument? my thinking was `Amount` is a value type and I don't see any behaviour on it that would be extended.